### PR TITLE
Gpfs path fix

### PR DESCRIPTION
**Pull request by hemalathagajendran**
## Pull request checklis

Please check the type of change your PR introduces:
- [ x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?
current functionality uses stat command to check whether mount path is gpfs or not

## What is the new behavior?
New functionality uses"/proc/mount" to check whether mount path is gpfs or not

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by hemalathagajendran** [comment]
Please use below images to test the fix.

quay.io/hemalatha_gajendran/operator_gpfspath_final:latest
quay.io/hemalatha_gajendran/driver_gpfspath_finalfix:latest

File: driver/csiplugin/nodeserver.go [hunk 1]
```diff
@@ -39,28 +39,68 @@ type ScaleNodeServer struct {
...
+        if !isGpfsPath {
+                return fmt.Errorf("checkGpfsType: the path [%s] is not a valid gpfs path ", strings.TrimPrefix(path, hostDir))
+        }
+
+        return nil
+}
+
+func getGpfsPaths(ctx context.Context) []string {
+        var gpfsPaths []string
+        gpfsPathCmd := `cat /proc/mounts | grep -i "gpfs"`
```
Comment (amdabhad) [inline driver/csiplugin/nodeserver.go:73]: Is it necessary to do case insensitive grep or `-i` can be removed from here?

File: driver/csiplugin/nodeserver.go [hunk 2]
```diff
@@ -39,28 +39,68 @@ type ScaleNodeServer struct {
...
+        cmd := exec.Command("bash", "-c", gpfsPathCmd)
+        output, err := cmd.CombinedOutput()
+        if err != nil {
+                klog.Errorf("[%s] Error in executing command: [%s]", utils.GetLoggerId(ctx), err)
+        }
```
Comment (amdabhad) [inline driver/csiplugin/nodeserver.go:78]: When there is an error, we don't need to process the output.., so next lines should be under else block or we should return from if bock

File: driver/csiplugin/nodeserver.go [hunk 3]
```diff
@@ -39,28 +39,68 @@ type ScaleNodeServer struct {
...
+                finalOutput := strings.Split(out, " ")
+                if len(finalOutput) == mountPathLength {
+                        if finalOutput[1] != "" && finalOutput[2] == "gpfs"{
+                                val,ok := os.LookupEnv(isOpenShiftCluster)
+                                if ok && val == "True"{
+                                        before, after, found := strings.Cut(finalOutput[1], "/var")
+                                        if found && before == hostDir && strings.HasPrefix(after,mountPath){
+                                                openShiftMountPath := before + after
+                                                gpfsPaths = append(gpfsPaths, openShiftMountPath)
+                                        }else{
```
Comment (amdabhad) [inline driver/csiplugin/nodeserver.go:91]: This else block seems unnecessary, is it required for any case?

Comment (hemalathagajendran) [reply driver/csiplugin/nodeserver.go:91]: yes this else block is required if we don't have "/var" in path then we should append the original path what we are getting from /proc/mounts. We cannot go with assumption from local clusters and among different ocp versions.

File: driver/csiplugin/nodeserver.go (hunk 1 above)
Comment (hemalathagajendran) [reply driver/csiplugin/nodeserver.go]: Done

File: driver/csiplugin/nodeserver.go (hunk 2 above)
Comment (hemalathagajendran) [reply driver/csiplugin/nodeserver.go]: Done

**Comment by hemalathagajendran** [comment]
Image for driver after addressing review comments: quay.io/hemalatha_gajendran/driver_gpfspath_fixwithcomments:latest

**Review by amdabhad — APPROVED** [review]

**Pull request merged**